File: domains/query-execution/steps/raw-analysis/user_review.py
"""7B: human review gate."""

from __future__ import annotations


def run(ctx):
    review = ctx.get("user_review")
    if review is None:
        card = {
            "formula": ctx.get("query_intent", {}).get("metric", "pending"),
            "event_set": ctx.get("query_intent", {}).get("event_set", []),
            "warnings": ctx.get("auto_review", {}).get("warnings", []),
        }
        return StepOutcome.await_input("raw.user_review", {"review_card": card})
    if review.get("status") == "changes":
        return StepOutcome.revise("changes", {"user_review": review}, "user requested changes")
    return StepOutcome.next({"user_review": review}, branch="confirmed")

File: domains/query-execution/steps/raw-analysis/user_review.ts
/** 7B: human review gate. */

import { StepOutcome } from "../../scheduler/scheduler.js";

export function run(ctx: Record<string, any>): StepOutcome {
  const review = ctx["user_review"];
  if (review === undefined || review === null) {
    const card = {
      formula: (ctx["query_intent"] ?? {})["metric"] ?? "pending",
      event_set: (ctx["query_intent"] ?? {})["event_set"] ?? [],
      warnings: (ctx["auto_review"] ?? {})["warnings"] ?? [],
    };
    return StepOutcome.await_input("raw.user_review", { review_card: card });
  }
  if (review["status"] === "changes") {
    return StepOutcome.revise("changes", { user_review: review }, "user requested changes");
  }
  return StepOutcome.next({ user_review: review }, "confirmed");
}
